experiments/profile_data.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+
+class ProfileData:
+    """ 
+    Class to hold and organize measurements for run configs
+    """
+
+    def __init__(self):
+        self._data = {}
+
+        self._best_config = None
+        self._best_measurement = None
+
+    def add_measurement(self, run_config, measurement):
+        """
+        Add the measurement for the given run_config
+        """
+        keys = self._extract_run_config_keys(run_config)
+        self._add_measurement_from_keys(keys, measurement)
+
+    def get_measurement(self, run_config):
+        """ 
+        Get the measurement belonging to a given run config
+        """
+        keys = self._extract_run_config_keys(run_config)
+        return self._get_measurement_from_keys(keys)
+
+    def get_config_count(self):
+        """ 
+        Get the total number of model configs in the data
+        """
+        count = 0
+        for ma_key in self._data.keys():
+            count += 1
+        return count
+
+    def get_measurement_count(self):
+        """ 
+        Get the total number of measurements in the data
+        """
+        count = 0
+        for ma_key in self._data.keys():
+            for pa_key in self._data[ma_key].keys():
+                count += 1
+        return count
+
+    def get_best_measurement(self):
+        """
+        Get the best overall measurement in the data
+        """
+        return self._best_measurement
+
+    def get_best_config(self):
+        """
+        Get the config corresponding to the best overall measurement in the data
+        """
+        return self._best_config
+
+    def _add_measurement_from_keys(self, keys, measurement):
+        self._update_best_trackers(keys, measurement)
+
+        curr_dict = self._data
+
+        for i in range(len(keys) - 1):
+            if keys[i] not in curr_dict:
+                curr_dict[keys[i]] = {}
+            curr_dict = curr_dict[keys[i]]
+        curr_dict[keys[-1]] = measurement
+
+    def _update_best_trackers(self, keys, measurement):
+        if not self._best_measurement or measurement > self._best_measurement:
+            self._best_measurement = measurement
+            self._best_config = keys[0]
```

Comment:
keys[0] = model_config - would be better to make to define this in the class and get rid of the magic number
also this will need to change to work for multi-model

See results.py for an example of how I handled this

Reply:
no longer using keys[0] or keys[1] anywhere.

Reply:
Yes, I see that, but I'm not sure what you've done is better/cleaner. You still need to have this implicit understanding that keys = {ma_key, pa_key} which is not documented anywhere outside of the method that actually creates the keys. 

It would be cleaner to just have ma_key and pa_key be separate variables and pass them in separately to the methods that need them.